app/controllers/admin/categories_controller.rb

```diff
@@ -25,12 +28,17 @@ def destroy
 
   def new_or_edit
     @categories = Category.find(:all)
-    @category = Category.find(params[:id])
+    #@category = Category.find(params[:id])
+    if @category_id == nil
```

Comment:
I originally had @category.id, but changed it in response to an error message that said I should use object_id instead of object.id